### PR TITLE
[SYCLomatic] Refine migration of CuDNN types

### PR DESCRIPTION
**Pull request by jzc**
Signed-off-by: Cai, Justin <justin.cai@intel.com>

### CODE REVIEW

**Review by 0x12CC — APPROVED** [review]

**Comment by ziranzha** [comment]
Please ask @intwanghao for approval.

File: clang/test/dpct/dnn/types.cu [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      // CHECK:      { // begin
+      // CHECK-NEXT: } // end
+  { // begin
+    from.handle = NULL;
+    from.handle = 0;
+    from.handle = nullptr;
+  } // end
+
+  // CHECK:      ~cudnnConvolutionDescriptorWrapper() {
+  // CHECK-NEXT: if (true) {
```
Comment (intwanghao) [inline clang/test/dpct/dnn/types.cu]: I think use operator= overload and user defined conversion function is a better solution.
Just like:
operator=(std::nullptr_t ptr){
 set the _q to null and clear worksapce
}
and 
operator bool(){
 return q != nullptr;
}

**Review by intwanghao — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc [hunk 2]
```diff
@@ -653,6 +662,16 @@ public:
                             _strides[i - 2];
     }
   }
+
+  convolution_desc &operator=(std::nullptr_t) {
+    return *this = convolution_desc();
+  }
+
+  operator bool() {
```
Comment (0x12CC) [inline clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc]:
```suggestion
  operator bool() const {
```

File: clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc [hunk 3]
```diff
@@ -198,6 +198,15 @@ public:
     }
     return result;
   }
+
+  operator bool() {
```
Comment (0x12CC) [inline clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc]:
```suggestion
  operator bool() const {
```

File: clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc [hunk 4]
```diff
@@ -2110,6 +2129,16 @@ public:
       size_t workspace_size, void *workspace);
 // DPCT_LABEL_END
 
+  operator bool() {
```
Comment (0x12CC) [inline clang/runtime/dpct-rt/include/dnnl_utils.hpp.inc]:
```suggestion
  operator bool() const {
```

**Review by 0x12CC — CHANGES_REQUESTED** [review]

**Review by 0x12CC — APPROVED** [review]

**Pull request merged**